druid-shell/src/platform/x11/window.rs

```diff
@@ -884,6 +884,31 @@ impl Window {
         ));
     }
 
+    fn set_cursor(&self, cursor: &Cursor) {
+        let cursors = &self.app.cursors;
+        #[allow(deprecated)]
+        let cursor = match cursor {
+            Cursor::Arrow => cursors.default,
+            Cursor::IBeam => cursors.text,
+            Cursor::Pointer => cursors.pointer,
+            Cursor::Crosshair => cursors.crosshair,
+            Cursor::OpenHand => {
+                warn!("Cursor::OpenHand not supported for x11 backend. using arrow cursor");
+                cursors.default
+            }
+            Cursor::NotAllowed => cursors.not_allowed,
+            Cursor::ResizeLeftRight => cursors.col_resize,
+            Cursor::ResizeUpDown => cursors.row_resize,
+            // TODO: (x11/custom cursor)
+            Cursor::Custom(_) => cursors.default,
+        };
+        let conn = self.app.connection();
+        let changes = ChangeWindowAttributesAux::new().cursor(cursor);
+        if let Err(e) = conn.change_window_attributes(self.id, &changes) {
+            error!("Changing window attributes failed {}", e);
```

Comment:
Maybe a cursor-specific error message would help in tracking things down? Especially if we use `change_window_attributes` for other attributes in the future...